uniprot-search/src/main/java/org/uniprot/store/search/domain2/UniProtKBSearchFields.java

```diff
@@ -0,0 +1,83 @@
+package org.uniprot.store.search.domain2;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.uniprot.cv.xdb.UniProtXDbTypes;
+import org.uniprot.store.config.common.FieldConfiguration;
+import org.uniprot.store.config.model.FieldItem;
+import org.uniprot.store.config.model.FieldType;
+import org.uniprot.store.config.uniprotkb.UniProtSearchFieldConfiguration;
+import org.uniprot.store.search.domain2.impl.SearchFieldImpl;
+
+public class UniProtKBSearchFields extends SearchFieldsLoader {
+    private FieldConfiguration configService;
+    private Set<SearchField> searchFields;
+    private Set<SearchField> sortFields;
+
+    public UniProtKBSearchFields() {
+        this.configService = UniProtSearchFieldConfiguration.getInstance();
+    }
+
+    @Override
+    public Set<SearchField> getSearchFields() {
+        if (this.searchFields == null) {
+
+            this.searchFields =
+                    this.configService.getAllFieldItems().stream()
+                            .filter(this::isSearchField)
+                            .map(fieldItem -> SearchFieldImpl.from(fieldItem))
```

Comment:
can use method ref

(the intellij sonarqube plugin is useful for spotting these suggestions. i use it)